cvmfs/download.cc

```diff
@@ -1844,11 +1848,21 @@ bool DownloadManager::ProbeHostsGeo() {
   for (unsigned i = 0; i < host_chain.size(); ++i)
     host_names.push_back(dns::ExtractHost(host_chain[i]));
   SortTeam(&host_names, &host_chain);
+
+  // add fallback proxy names to the end of the host list
+  unsigned first_geo_fallback = host_names.size();
+  for (unsigned i = fallback_group; i < proxy_chain.size(); ++i) {
+    // note that there's only ever one fallback proxy name per group
+    host_names.push_back(proxy_chain[i][0].host.name());
+  }
+  // There's no reason to sort fallbacks, they're set by a widely shared config
```

Comment:
When I wrote that I thought it would be significant extra work to sort, because it would require also sorting the corresponding subset of the proxy chain while not sorting the pre-fallback part of the proxy chain.   Now looking back I guess it would just require one more step of extracting the fallback part of the proxy chain.  Do you think I should do it?

Thanks for making me look at this again because now I realize that there's a problem below.  I'll add my own note in the appropriate place.